include/pybind11/pytypes.h

```diff
@@ -2327,6 +2333,20 @@ bool object_api<D>::rich_compare(object_api const &other, int value) const {
     return rv == 1;
 }
 
+inline std::nested_exception error_fetch_and_normalize::get_cause_as_nested() const {
+    auto cause = reinterpret_steal<object>(PyException_GetCause(m_value.ptr()));
+    if (not cause or cause.is_none())
+        return std::nested_exception();
+    auto typ = type::of(cause);
+    auto tb = reinterpret_steal<object>(PyException_GetTraceback(cause.ptr()));
+    PyErr_Restore(typ.release().ptr(), cause.release().ptr(), tb.release().ptr());
+    try {
+        throw error_already_set();
```

Comment:
Throwing is very slow and something that is best avoided here. is there a way we avoid having to throw the exception just to catch and return an empty std::nested_exception?